src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1241,10 +1580,10 @@ fn anchor_failure(
     });
 }
 
-fn ambiguity_error(
+fn ambiguity_error<'a>(
     cx: &DocContext<'_>,
     item: &Item,
-    path_str: &str,
+    path_str: &'a str,
```

Comment:
This can get rid of the lifetime parameter now that it returns `ErrorKind<'static>`.